File: pyg_parent/pyg_manager_web_91/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.security.access.method.P;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage/{page}/{size}")
    public PageResult<TbBrand> findPage(@PathVariable("page") int page, @PathVariable("size") int size){
        return brandService.findPage(page, size);
    }

    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand brand){
        try{
            brandService.save(brand);
            //成功，返回成功
            return new Result(true, "保存成功！");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false, "保存失败！");
        }
    }

    @RequestMapping("/findOne/{id}")
    public TbBrand findOne(@PathVariable("id") Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/delete/{ids}")
    public Result delete(@PathVariable("ids") String ids){
        try{
            brandService.delete(ids);
            //成功，返回成功
            return new Result(true, "删除成功！");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false, "删除失败！");
        }
    }
}

File: pyg_parent/pyg_manager_web_91/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope, $http, brandService, $controller) {
    $controller('baseController',{$scope:$scope});
    $scope.findAll = function () {
        //发送请求查询所有品牌，展示到列表
        brandService.findAll().success(function (res) {
            $scope.brandList = res;
        })
    }

    $scope.findPage=function(page,size){
        brandService.findPage(page, size).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

    $scope.entity = {};//对应后台接受的实体类
    $scope.save = function () {
        //发送保存请求，接受返回值并重新刷新界面
        brandService.save($scope.entity).success(function (res) {
            //提示客户
            alert(res.message);
            //如果成功，刷新界面
            if(res.success){
                $scope.reloadList();
            }
        })
    }
    $scope.findOne = function (id) {
        //根据主键查询单个品牌对象，将对象复制给entity模型
        brandService.findOne(id).success(function (res) {
            $scope.entity = res;
        });
    }

    $scope.delete = function () {
        //1.发送删除请求
        brandService.delete($scope.selectIds).success(function (res) {
            //2.接口返回值，提示；判断如果成功，刷新界面，清空选中项数组
            alert(res.message);
            if(res.success){
                $scope.reloadList();
                $scope.selectIds = [];
            }
        })
    }
});